### PR TITLE
#94/display gmaps routes 3

### PR DESCRIPTION
**Pull request by ferdousara-parvin**
What I was able to achieve for this PR: Create a DirectionsResult object in the RoutesActivityViewModel which contains all the necessary information related to the possible routes from an origin to a destination with a provided transport type.

Even though user story #94 is not done (UI part left), a good chunk is done so I thought I would merge it as some of you guys will need the information retrieved from the Google Maps Directions API soon. 

Important notes:

- Don't be scared by the number of files changed, most of them are classes from Google Maps Services library so you can disregard them.

- The important files you should review are: DirectionsApiDataParser, DirectionsApiDataRetrieval, RoutesActivity, ClassConstants and RoutesActivityViewModel files.

- There's a folder called GoogleMapsServicesTools. DO NOT change anything in these files. The two folders inside are essentially a duplicate of "model" and "internal" folder in the Google Maps Services library. I needed to create this duplication because or else the project wasn't building properly.

- Instead of using the Google Maps services library, I had to import it into our own code to be able to modify some of the models that I needed from the library because the mapping from the JSON string returned by the API and the models provided by Google maps were not identical so Gson was not working properly.

### CODE REVIEW

**Comment by thomasgauvin** [comment]
@ferdousara-parvin could you add a comment indicating which files were mostly autogenerated to save time for the pr reviewers?

File: app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivity.java [hunk 1]
```diff
@@ -76,6 +94,20 @@ public boolean onKeyDown(int keyCode, KeyEvent event)  {
         return super.onKeyDown(keyCode, event);
     }
 
+    public void directionsApiCallBack(DirectionsResult result)
+    {
+        mViewModel.setDirectionsResult(result);
+        System.out.println("lol");
```
Comment (thomasgauvin) [inline app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivity.java]: @ferdousara-parvin

Comment (thomasgauvin) [reply app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivity.java]: :')

File: app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivityViewModel.java [hunk 2]
```diff
@@ -35,4 +40,39 @@ public void setFrom(Place from) {
...
+     * Build the URL that will be used to call the Google Maps Directions API by passing the necessary parameters
+     *
+     * @param from: latitude and longitude of the origin
+     * @param to: latitude and longitude of the destination
+     * @param transportType: main transport type to get from origin to destination
+     *
+     */
+    public String buildUrl(LatLng from, LatLng to, @ClassConstants.TransportType String transportType) {
+        // Origin of route
+        String str_origin = "origin=" + from.latitude + "," + from.longitude;
```
Comment (thomasgauvin) [inline app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivityViewModel.java]: I think all these comments are useless, the naming of your variables is good enough

**Comment by thomasgauvin** [comment]
Hey Ferdou,

Since this PR is quite big, could you actually indicate which files the reviewers should look at? It is difficult to browse the files and know which ones need reviewing

File: app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivity.java (hunk 1 above)
Comment (ferdousara-parvin) [reply app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivity.java]: smh thanks!

File: app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivityViewModel.java (hunk 2 above)
Comment (ferdousara-parvin) [reply app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivityViewModel.java]: you're right!

**Comment by ferdousara-parvin** [comment]
Also, I am aware that my commit messages don't follow the format we agreed on (having the issue number at the beginning of the commit message). I'll be more mindful of that in my next commits! Sorry!

File: app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivity.java [hunk 3]
```diff
@@ -96,6 +126,17 @@ private void openSearchPage(){
...
+        Coordinates toCoordinates = (mViewModel.getTo().getCenterCoordinates());
+        to = new MarkerOptions().position(new LatLng(toCoordinates.getLatitude(), toCoordinates.getLongitude()));
+
+    }
+
+    private void setTo() {
+        Coordinates fromCoordinates = (mViewModel.getFrom().getCenterCoordinates());
+        from = new MarkerOptions().position(new LatLng(fromCoordinates.getLatitude(), fromCoordinates.getLongitude()));
+    }
+
```
Comment (CarenRizk) [inline app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivity.java]: Maybe I don't understand this right.. but why in the setFrom() you are setting the to, and vice versa for the setTo() ?

Comment (ferdousara-parvin) [reply app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivity.java]: oh oups I made a mistake here, thanks for catching it :)

**Pull request merged**